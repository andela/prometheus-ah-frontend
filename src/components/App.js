import React, { Component } from 'react';

/**
 * @class App
 */
class App extends Component {
  render() {
    return (
      <div>
        <h1>Welcome to Authors haven</h1>
      </div>
    )
  }
}

export default App;
