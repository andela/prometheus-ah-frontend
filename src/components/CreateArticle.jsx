import React, { Component } from 'react';

/**
 * @class App
 */
class CreateArticle extends Component {
  /**
   * @description Render the JSX template
   *
   * @returns {JSX} render JSX template
   *
   * @memberof Header
   */
  render() {
    return (
      <div>
        <h1>Create Article page</h1>
      </div>
    );
  }
}

export default CreateArticle;
