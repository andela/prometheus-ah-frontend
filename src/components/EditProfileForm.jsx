import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import TextField from './common/TextField';
import TextArea from './common/TextArea';
import ErrorAlertNotification from './common/ErrorAlertNotification';

import EditProfileValidate from '../middlewares/EditProfileValidate';

/**
 * @EditProfileForm page
 */
export class EditProfileForm extends Component {
  state = {
    image: '',
    firstname: '',
    lastname: '',
    username: '',
    bio: '',
    profileImage: null,
    errors: {},
  };

  /**
   * componentDidMount
   *@param {object} event
   * @return {void}
   */
  componentDidMount() {
    const { profile } = this.props;
    this.setState({
      image: profile.image,
      firstname: profile.firstname,
      lastname: profile.lastname,
      bio: profile.bio,
      username: profile.username,
    });
  }

  /**
   * onChange
   *@param {object} event
   * @return {void}
   */
  onChange = (event) => {
    const { errors } = this.state;
    if (errors[event.target.name]) {
      const newErrors = Object.assign({}, errors);
      delete newErrors[event.target.name];
      this.setState({
        [event.target.name]: event.target.value,
        errors: newErrors
      });
    } else {
      this.setState({
        [event.target.name]: event.target.value
      });
    }
  }

  /**
   * onSubmit
   *@param {object} event
   * @return {void}
   */
  onSubmit = (event) => {
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      const { editProfile, history } = this.props;
      const {
        image, firstname, lastname, bio, profileImage, username
      } = this.state;
      const user = {
        image,
        firstname,
        lastname,
        bio,
        profileImage,
        username
      };
      editProfile(user, history.push('/profiles'));
    }
  }

  /**
   * onSubmit
   *@param {object} event
   * @return {void}
   */
  fileChangedHandler = (event) => {
    this.setState({ profileImage: event.target.files[0] });
    if (event.target.files && event.target.files[0]) {
      const reader = new FileReader();
      reader.onload = (e) => {
        $('#myimage').attr('src', e.target.result); //eslint-disable-line
      };
      reader.readAsDataURL(event.target.files[0]);
    }
  }

  /**
   * isValid
   * @returns {*} - state
   */
  isValid() {
    const { errors, isValid } = EditProfileValidate.InputValidation(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  /**
   * render
   *
   * @return {void}
   */
  render() {
    const {
      errors, image, firstname, lastname, bio
    } = this.state;

    const { error } = this.props;

    return (
      <form onSubmit={this.onSubmit} className="form-horizontal">
        <div className="panel panel-default">
          {
            (error && error.message) ? <ErrorAlertNotification errors={error.message} /> : ''
          }
          <div className="panel-body">
            <div className="form-group">
              <div className="container profile-img form-group text-center">
                <img id="myimage" src={image} className="img-circle profile-avatar image"
                  alt="User avatar"
                />
                <div className="overlay">
                  <button type="button" className="icon" title="User Profile"
                    onClick={() => this.fileInput.click()}
                  >
                    <i className="fas fa-camera-retro" />
                  </button>
                </div>
              </div>
              <input className="hideInput" type="file" name="profileImage"
                id="profileImage" onChange={this.fileChangedHandler}
                ref={(fileInput) => { this.fileInput = fileInput; }} accept="image/*"
              />
            </div>
            <div className="form-group">
              <label htmlFor="firstname">First Name</label>
              <TextField
                error={errors.firstname}
                label="First Name"
                onChange={this.onChange}
                value={firstname}
                field="firstname"
                placeholder="enter your first name"
                id="firstname"
              />
            </div>
            <div className="form-group">
              <label htmlFor="lastname">Last Name</label>
              <TextField
                error={errors.lastname}
                label="Last Name"
                onChange={this.onChange}
                value={lastname}
                field="lastname"
                placeholder="enter your last name"
                id="lastname"
              />
            </div>
            <div className="form-group">
              <label htmlFor="bio">Bio</label>
              <TextArea
                error={errors.bio}
                label="Bio"
                onChange={this.onChange}
                value={bio}
                field="bio"
                placeholder="enter bio here..."
              />
            </div>
          </div>
          <Link to="{routes.PROFILE_PAGE}">
            <button className="btn btn-secondary" type="button">Cancel</button>
          </Link>
          &nbsp;
          <button id="register" className="btn btn-danger" type="submit">Save Changes</button>
        </div>
      </form>
    );
  }
}

EditProfileForm.propTypes = {
  profile: PropTypes.shape({
  }).isRequired,
  editProfile: PropTypes.func.isRequired,
  error: PropTypes.shape({
  }),
  history: PropTypes.shape({
  }).isRequired,
};

const mapStateToProps = state => ({
  error: state.editProfile.error
});

export default connect(mapStateToProps, null)(EditProfileForm);
