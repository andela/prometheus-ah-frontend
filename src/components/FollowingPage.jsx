import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import UserNavigation from './navigation/UserNavigation';
import {
  getMyFollowingActions,
  getMyFollowersActions
} from '../actions/followAction';

/**
 * @description class to display followers and following
 *
 * @class DisplayFollow
 *
 * @extends {Component}
 */
class DisplayFollow extends Component {
  state = {
    selected: ''
  }

  /**
   * @description Render the JSX template
   *
   * @memberof DisplayFollow
   * @returns {JSX} render JSX template
   */
  componentWillMount() {
    const { myFollowers, myFollowing, user } = this.props;
    /* myFollowers(user); */
    myFollowing(user);
  }

  renderFollows = follows => follows
    && follows.map(
      follow => <div key={follow.username}>{follow.username}</div>
    );

  /**
   *
   * @param {*} event
   * @returns {*} - state
   */
  render() {
    console.log('foloowers', this.props.followingData);
    const {
      isFollowing,
      followingData: { myTotalFollow, authorsIFollow }
    } = this.props;
    return (
      <Fragment>
        <h1>CHIDINMAAAA</h1>
        {/* <Header /> */}
        <UserNavigation />
        <div className="card text-center" style={{ width: '50%', margin: 'auto', boxShadow: '2px 6px' }}>
          <div className="card-body">
            <h5 className="card-title">
            Followings
              <br />
              <span>{myTotalFollow}</span>
              <div>{this.renderFollows(authorsIFollow)}</div>
            </h5>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  followingData: state.followReducer.following,
  user: state.auth.user.username
});

const mapDispatchToProps = dispatch => ({
  // myFollowers: (username) => {
  //   dispatch(getMyFollowersActions(username));
  // },
  myFollowing: (username) => {
    dispatch(getMyFollowingActions(username));
  }
});
export default
connect(
  mapStateToProps,
  mapDispatchToProps
)(DisplayFollow);
