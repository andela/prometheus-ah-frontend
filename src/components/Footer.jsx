import React from 'react';

const Footer = () => (
  <div className="footer text-center">
    <h4>Authors Haven</h4>
    <p>Copyright 2018 &copy; All Rights Reserved. Authors Haven Inc.</p>
    <span className="circle">
      <a href="/"><i className="fab fa-facebook-f fab-color" /></a>
    </span>
    &nbsp;
    <span className="circle">
      <a href="/"><i className="fab fa-twitter fab-color" /></a>
    </span>
    &nbsp;
    <span className="circle">
      <a href="/"><i className="fab fa-instagram fab-color" /></a>
    </span>
    &nbsp;
    <span className="circle">
      <a href="/"><i className="fab fa-google-plus-square fab-color" /></a>
    </span>
  </div>
);

export default Footer;
