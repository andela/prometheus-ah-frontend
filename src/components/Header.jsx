import React from 'react';

const Header = () => (
  <nav className="navbar navbar-expand-sm bg-dark navbar-dark">
    <a className="navbar-brand" href="/">Authors Haven</a>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#collapsibleNavbar">
      <span className="navbar-toggler-icon" />
    </button>
    <div className="collapse navbar-collapse" id="collapsibleNavbar">
      <ul className="navbar-nav">
        <li className="nav-item">
          <a className="nav-link" href="/">Home</a>
        </li>
      </ul>
    </div>
  </nav>
);

export default Header;
