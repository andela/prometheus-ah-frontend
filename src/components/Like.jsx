import React from 'react';
import PropTypes from 'prop-types';

/**
 * function Like
 * @returns {Object} Like component
 */
const Like = ({
  iconClassName, spanClassName, count, ...rest
}) => (
  <i className={iconClassName} {...rest}>
    <span className={spanClassName}>
      {' '}
      {count}
    </span>
  </i>
);
Like.propTypes = {
  iconClassName: PropTypes.string,
  spanClassName: PropTypes.string,
  count: PropTypes.number
};

Like.defaultProps = {
  iconClassName: '',
  spanClassName: ''
};

export default Like;
