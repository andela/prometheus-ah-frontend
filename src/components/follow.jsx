import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  followUserAction, unfollowUserAction, getMyFollowingActions, getMyFollowersActions
} from '../actions/followAction';

/**
 * @description follow button class allows users follow one another
 * @returns {*} jsx
 */
class FollowButton extends Component {
  /**
   *
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      isFollowing: false
    };
  }

  componentWillMount() {
    this.props.myFollowers(this.props.username);
    this.props.myFollowing(this.props.username);
  }

handleClick = () => {
  if (this.props.isFollowing === true) {
    this.props.unfollowUser(this.props.username, this.props.user);
  } else {
    this.props.followUser(this.props.username, this.props.user);
  }
}

renderFollows = follows => follows && follows.map(follow => <span key={follow.username}>{follow.username}</span>)

/**
   *
   * @param {*} event
   * @returns {*} - state
   */
render() {
  const { isFollowing, followingData: { myTotalFollow, authorsIFollow } } = this.props;
  return (
    <div className="container">
      <button
        type="button"
        className="btn btn-outline-danger"
        onClick={this.handleClick}
      >
        {isFollowing ? 'Unfollow' : 'Follow'}

      </button>
      <div>
        <span>{myTotalFollow}</span>
        {this.renderFollows(authorsIFollow)}
      </div>
    </div>
  );
}
}

const mapStateToProps = state => ({
  isFollowing: state.followReducer.isFollowing,
  error: state.followReducer.error,
  followingData: state.followReducer.following,
  user: state.auth.user.username
});
const mapDispatchToProps = dispatch => ({
  followUser: (username, user) => {
    dispatch(followUserAction(username, user));
  },
  unfollowUser: (username, user) => {
    dispatch(unfollowUserAction(username, user));
  },
  myFollowers: (username) => {
    dispatch(getMyFollowersActions(username));
  },
  myFollowing: (username) => {
    dispatch(getMyFollowingActions(username));
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(FollowButton);
