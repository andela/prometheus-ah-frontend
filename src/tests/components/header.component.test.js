import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let wrapper;

describe('Header Component', () => {
  beforeEach(() => {
    const props = {
      logout: jest.fn(),
      showModal: jest.fn(),
      showModalSignup: jest.fn()
    };

    wrapper = shallow(<Header {...props} />);
  });

  it('should render the header component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GuestNavigation').exists()).toBe(true);
    expect(wrapper.find('UserNavigation').exists()).toBe(false);
  });

  it('should render a component', () => {
    const props = {
      auth: true,
      logout: jest.fn(),
      showModal: jest.fn(),
      showModalSignup: jest.fn()
    };
    const wrapper1 = shallow(<Header {...props} />);

    expect(wrapper1).toMatchSnapshot();
    expect(wrapper1.find('GuestNavigation').exists()).toBe(false);
    expect(wrapper1.find('UserNavigation').exists()).toBe(false);
  });
});
