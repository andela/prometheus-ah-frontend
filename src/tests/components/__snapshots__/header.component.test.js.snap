// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render a component 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark navbar-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Authors Haven
  </a>
  <button
    className="navbar-toggler"
    data-target="#collapsibleNavbar"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="collapsibleNavbar"
  >
    <ul
      className="navbar-nav"
    >
      <Connect(UserNavigation) />
    </ul>
  </div>
</nav>
`;

exports[`Header Component should render the header component correctly 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark navbar-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Authors Haven
  </a>
  <button
    className="navbar-toggler"
    data-target="#collapsibleNavbar"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="collapsibleNavbar"
  >
    <ul
      className="navbar-nav"
    >
      <GuestNavigation
        showLoginModal={[MockFunction]}
        showSignupModal={[MockFunction]}
      />
    </ul>
  </div>
</nav>
`;
